docs/conf.py

```diff
@@ -123,6 +123,8 @@
 # If true, `todo` and `todoList` produce output, else they produce nothing.
 todo_include_todos = False
 
+#Forces order of functions to appear as in source code, rather than sorted.
+autodoc_member_order = 'bysource'
```

Comment:
this was done so semigroups are listed before monoids (alex also added these lines so that his stuff would be sorted in the correct order)